File: app/src/main/java/com/lambdaschool/congressdata/OfficialOverview.java
package com.lambdaschool.congressdata;

import com.lambdaschool.congressdataapiaccess.CongresspersonOverview;

public class OfficialOverview {
    private String firstName, middleName, lastName;
    private String party, state;
    private String displayName;
    private String id;

    public OfficialOverview(String firstName, String middleName, String lastName, String party, String state, String id) {
        this.firstName = firstName;
        this.middleName = middleName;
        this.lastName = lastName;
        this.party = party;
        this.state = state;
        this.displayName = buildDisplayName();
        this.id = id;
    }

    public OfficialOverview(CongresspersonOverview congresspersonOverview) {
        this.firstName = congresspersonOverview.getFirstName();
        this.middleName = congresspersonOverview.getMiddleName();
        this.lastName = congresspersonOverview.getLastName();
        this.party = congresspersonOverview.getParty();
        this.state = congresspersonOverview.getState();
        this.displayName = buildDisplayName();
        this.id = congresspersonOverview.getId();
    }

    private String buildDisplayName() {
        StringBuilder nameBuilder = new StringBuilder();
        nameBuilder.append(firstName);
        if (!middleName.equals("null")) {
            nameBuilder.append(middleName).append(" ");
        }
        nameBuilder.append(lastName);
        return nameBuilder.toString().toLowerCase();
    }

    public String getDisplayName() {
        return displayName;
    }

    public String getId() {
        return id;
    }

    public String getParty() {
        return party;
    }

    public String getState() {
        return state;
    }
}

File: app/src/main/java/com/lambdaschool/congressdata/OfficialOverview.kt
package com.lambdaschool.congressdata

import com.lambdaschool.congressdataapiaccess.CongresspersonOverview

class OfficialOverview {
    private var firstName: String? = null
    private var middleName: String? = null
    private var lastName: String? = null
    var party: String? = null
        private set
    var state: String? = null
        private set
    var displayName: String? = null
        private set
    var id: String? = null
        private set

    constructor(firstName: String? = null, middleName: String? = null, lastName: String? = null, party: String? = null, state: String? = null, id: String? = null) {
        this.firstName = firstName
        this.middleName = middleName
        this.lastName = lastName
        this.party = party
        this.state = state
        this.displayName = buildDisplayName()
        this.id = id
    }

    constructor(congresspersonOverview: CongresspersonOverview) {
        this.firstName = congresspersonOverview.firstName
        this.middleName = congresspersonOverview.middleName
        this.lastName = congresspersonOverview.lastName
        this.party = congresspersonOverview.party
        this.state = congresspersonOverview.state
        this.displayName = buildDisplayName()
        this.id = congresspersonOverview.id
    }

    private fun buildDisplayName(): String {
        val nameBuilder = StringBuilder()
        nameBuilder.append(firstName).append(" ")
        if (middleName != null) {
            nameBuilder.append(middleName).append(" ")
            nameBuilder.append(lastName)
        } else {
            nameBuilder.append(lastName)
        }

        return nameBuilder.toString().toLowerCase()
    }
}
